### PR TITLE
Make strings in `init_builtin_classes` function `const`

### PR DESCRIPTION
**Pull request by yuk1ty**
#266 

Replaced hard-coded strings with `const`. I'll reorganize the order of `const`s in another pull request 👍🏻

@mtshiba

### CODE REVIEW

File: compiler/erg_compiler/context/initialize/classes.rs [hunk 1]
```diff
@@ -1148,227 +1198,221 @@ impl Context {
...
-            poly("Map", vec![ty_tp(T.clone())]),
+            poly(MAP, vec![ty_tp(T.clone())]),
             map,
             Private,
             Const,
-            Some("map"),
+            Some(FUNC_MAP),
         );
         self.register_builtin_type(
-            poly("Revered", vec![ty_tp(T.clone())]),
```
Comment (yuk1ty) [inline compiler/erg_compiler/context/initialize/classes.rs:1303]: Just a quick confirmation, is this a typo of "Reversed"? 🤔

Comment (mtshiba) [reply compiler/erg_compiler/context/initialize/classes.rs:1303]: Oh, yes. "Reversed" is right.

Comment (yuk1ty) [reply compiler/erg_compiler/context/initialize/classes.rs:1303]: Thanks 😊

**Comment by mtshiba** [comment]
Thank you!

**Pull request merged**